miden-lib/src/wallets/mod.rs

```diff
@@ -0,0 +1,63 @@
+use crate::assembler::assembler;
+use miden_objects::{
+    accounts::{Account, AccountCode, AccountId, AccountStorage, AccountType, AccountVault},
+    assembly::ModuleAst,
+    crypto::merkle::MerkleStore,
+    utils::vec,
+    AccountError, Word, ZERO,
+};
+
+use rand::Rng;
+
+pub enum AuthScheme {
+    RpoFalcon512 { pub_key: Word }, // we'll later change this to `PublicKey` from Falcon module.
+}
+
+/// Creating accounts with the basic wallet code, the authentication scheme can be defined.
+pub fn create_basic_wallet(auth_scheme: AuthScheme) -> Result<(Account, Word), AccountError> {
+    let mut rng = rand::thread_rng();
+    let mut bytes: [u8; 32] = [0; 32];
+    rng.fill(&mut bytes);
```

Comment:
Since we use random value generator here, this will need to be gated on `std`.

Another option: we can pass in `init_seed` as a parameter and then do the grinding internally. The method then would look like this:
```Rust
pub fn create_basic_wallet(
    auth_scheme: AuthScheme,
    init_seed: [u8; 32],
) -> Result<(Account, Word), AccountError>
```

Reply:
Also, looking at the rest of the code, should we pass in other parameters into this? e.g., for on-chain vs. off-chain.

Reply:
jup, when we have `on-chain` I would pass it on. Atm it doesn't make sense, right?

Reply:
Yep - for now we don't need it.